capa/features/extractors/cape/global_.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (C) 2020 Mandiant, Inc. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at: [package root]/LICENSE.txt
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+#  is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+import logging
+from typing import Tuple, Iterator
+
+from capa.features.address import Address, NO_ADDRESS
+from capa.features.common import (
+    OS,
+    OS_ANY,
+    ARCH_I386,
+    ARCH_AMD64,
+    ARCH_ANY,
+    FORMAT_PE,
+    FORMAT_ELF,
+    FORMAT_UNKNOWN,
+    OS_WINDOWS,
+    OS_LINUX,
+    Arch,
+    Format,
+    Feature,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+def guess_elf_os(file_output) -> Iterator[Tuple[Feature, Address]]:
+    # operating systems recognized by the file command: https://github.com/file/file/blob/master/src/readelf.c#L609
+    if "Linux" in file_output:
+        return OS(OS_LINUX), NO_ADDRESS
+    elif "Hurd" in file_output:
+        return OS("hurd"), NO_ADDRESS
+    elif "Solairs" in file_output:
+        return OS("solaris"), NO_ADDRESS
+    elif "kFreeBSD" in file_output:
+        return OS("freebsd"), NO_ADDRESS
+    elif "kNetBSD" in file_output:
+        return OS("netbsd"), NO_ADDRESS
+    else:
+        return OS(OS_ANY), NO_ADDRESS
+
+
+def extract_arch(static) -> Iterator[Tuple[Feature, Address]]:
+    if "Intel 80386" in static["target"]["type"]:
+        return Arch(ARCH_I386), NO_ADDRESS
+    elif "x86-64" in static["target"]["type"]:
+        return Arch(ARCH_AMD64), NO_ADDRESS
+    else:
```

Comment:
perhaps warn here so we are notified to update our logic?

Reply:
do you mean with regards to the use of NO_ADDRESS? isn't that going to be the case for Arch and OS features?

Reply:
no, that we didn't recognize the architecture
```suggestion
    else:
        logger.warn("unrecognized arch: %s", static["file"]["type"])
```